x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.js

```diff
@@ -53,6 +53,51 @@ class LinksMenuUI extends Component {
     };
   }
 
+  openInDiscover = async () => {
+    const {
+      services: { share },
+    } = this.props.kibana;
+    const discoverLocator = share.url.locators.get('DISCOVER_APP_LOCATOR');
+    const job = mlJobService.getJob(this.props.anomaly.jobId);
+
+    const index = job.datafeed_config.indices[0];
+    const dataViewId = (await getDataViewIdFromName(index)) || index;
```

Comment:
I think you might have uncovered a bug in `getDataViewIdFromName`. 

Create two data views:
- cloudwatch-2019.*
- cloudwatch-2019.11.04

The create a job off the wildcard index pattern `cloudwatch-2019.*`. The new Discover action for my setup then opens Discover in the wrong data view `cloudwatch-2019.11.04`.

Reply:
Yes, this is a bug.
The new dataviews service isn't as accurate as the old one. In other places I've fixed this by performing a second find on the results. e.g.
https://github.com/elastic/kibana/blob/c6d41d07d727c1257f482880f22564bae30449fb/x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts#L628

I'll create a PR for this as there is another place this needs to be fixed.
https://github.com/elastic/kibana/pull/118006#discussion_r756187254

Reply:
Raised https://github.com/elastic/kibana/pull/120737 to fix this

Reply:
Thanks @jgowdyelastic for the quick turn around 🙏  I merged the main branch your latest changes and the fix works great.

Reply:
Confirmed that the target data view is set correctly with the changes merged in from https://github.com/elastic/kibana/pull/120737.